### PR TITLE
Fix for parallel test results

### PR DESCRIPTION
**Pull request by Ameenasuhani**
# Description
Fixes the parallel module to indicate failed test case

issue logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-4TY4ZQ/
Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-0RZM0B/

### CODE REVIEW

File: tests/parallel/test_parallel.py [hunk 1]
```diff
@@ -28,13 +28,16 @@ def run(**kwargs):
...
+    results = []
     parallel_tests = kwargs["parallel"]
 
     with parallel() as p:
         for test in parallel_tests:
             p.spawn(execute, test, kwargs, results)
+
+    for result in results:
+        if not result:
+            return 1
```
Comment (psathyan) [inline tests/parallel/test_parallel.py:40]: I think we could do
```
test_rc = 0
for key, value in results:
  log.info(f"{key} test result is {PASS if value == 0 else FAILED}")
  if value != 0: test_rc = value

return test_rc
```
This should suffice.

**Review by sunilkumarn417 — APPROVED** [review]
LGTM

File: tests/parallel/test_parallel.py [hunk 2]
```diff
@@ -21,24 +21,35 @@
...
 """
-
 import importlib
+import logging
+from time import sleep, time
 
 from ceph.parallel import parallel
 
+logger = logging.getLogger(__name__)
+log = logger
```
Comment (psathyan) [inline tests/parallel/test_parallel.py:31]: We could do `log = logging.getLogger(__name__)`

**Comment by Ameenasuhani** [comment]
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-KYNKVF/Parallel_run_0.log
cephadm-adopt is failing as it is not able to find /etc/ceph folder on grafana node  and rbd i/o is failing because orch command is being used before the adoption

**Review by psathyan — APPROVED** [review]

**Pull request merged**